### PR TITLE
Pwa 1471

### PR DESCRIPTION
**Pull request by pedrochiossi**
## Description
PR to add functionality to parse media query CSS rules that are set up in pageBuilder and apply their respective styles to the following content types:

- Row
- Column
- Slider
- Banner
- Tabs
- Tab Item

## Related Issue

Closes PWA-1471

## Acceptance

PWA storefront should be able to render styles set with media query breakpoints configured in Page Builder for the content types mentioned above.

#### Test scenario(s) for direct fix/feature

1. Verify desktop and device specific breakpoints are configured in Page Builder
2. Verify the PWA storefront renders media query breakpoints for desktop and devices for multiple Page Builder content types: Rows, Columns, Tabs, Tab Items, Banners, Sliders, and Slides
3. Verify that all existing cypress snapshots tests are passing
4. Verify that all media query cypress snapshots tests are passing
#### Is Browser/Device testing needed?
No

#### Any ad-hoc/edge case scenarios that need to be considered?
No

## Checklist

- [ ]  I have added tests to cover my changes, if necessary.
- [ ]  I have added translations for new strings, if necessary.
- [ ]  I have updated the documentation accordingly, if necessary.

### CODE REVIEW

File: packages/peregrine/lib/hooks/useMediaQuery.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    const [styles, setStyles] = useState({});
+
+    const { mediaQueries } = props;
+
+    useEffect(() => {
+        if (!mediaQueries) return;
+
+        const mqlList = mediaQueries.map(({ media }) => matchMedia(media));
+
+        const handleMatch = (query, i) => {
```
Comment (jcalcaben) [inline packages/peregrine/lib/hooks/useMediaQuery.js:25]: can we add unit tests that cover both cases in this logic?

**Review by jcalcaben — CHANGES_REQUESTED** [review]

File: venia-integration-tests/cypress.config.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
-    "baseUrl": "https://develop.pwa-venia.com/",
+    "baseUrl": "https://magento-venia-concept-2tvqr.local.pwadev:9047/",
```
[the remark this answers is not in the document]
Comment (pedrochiossi) [reply venia-integration-tests/cypress.config.json:2]: Yes, my bad, pushed that by mistake.

File: packages/pagebuilder/lib/utils.js [hunk 3]
```diff
@@ -173,3 +173,49 @@ export function getIsHidden(node) {
...
+
+    return result;
+}
+
+/**
+ *
+ * @param  node
+ * @returns {{mediaQueries: [{media: string, style: string}]}}
+ */
+export function getMediaQueries(node) {
```
[the remark this answers is not in the document]
Comment (pedrochiossi) [reply packages/pagebuilder/lib/utils.js:201]: Yes, forgot about those helpers.

File: packages/pagebuilder/lib/utils.js [hunk 4]
```diff
@@ -173,3 +173,49 @@ export function getIsHidden(node) {
...
     };
 }
+
+/**
+ * Converts a CSS string style into a JSX object inline style
+ *
+ * @param {String} style
+ * @returns {Object}
+ */
+export function cssToJSXStyle(style) {
```
[the remark this answers is not in the document]
Comment (pedrochiossi) [reply packages/pagebuilder/lib/utils.js:183]: Yes, I'm on it.

File: packages/peregrine/lib/hooks/useMediaQuery.js (hunk 1 above)
Comment (pedrochiossi) [reply packages/peregrine/lib/hooks/useMediaQuery.js:25]: Ok, I can try mocking unmatched queries.

**Review by jcalcaben — DISMISSED** [review]

File: packages/pagebuilder/lib/ContentTypes/Banner/banner.js [hunk 5]
```diff
@@ -92,7 +96,8 @@ const Banner = props => {
         borderColor,
         borderWidth,
         borderRadius,
-        textAlign
+        textAlign,
+        ...(appearance !== 'poster' && mediaQueryStyles)
```
Comment (mikhaelbois) [inline packages/pagebuilder/lib/ContentTypes/Banner/banner.js:100]: Why wouldn't we want other banner types to have the mobile styles?

File: packages/pagebuilder/lib/ContentTypes/Column/column.js [hunk 6]
```diff
@@ -219,6 +224,12 @@ Column.propTypes = {
     marginLeft: string,
     marginRight: string,
     marginTop: string,
+    MediaQueries: arrayOf(
```
Comment (mikhaelbois) [inline packages/pagebuilder/lib/ContentTypes/Column/column.js:227]: Should be `mediaQueries`.

**Review by mikhaelbois — COMMENTED** [review]
@pedrochiossi 
Left comments on two files.
Cypress tests ✅ 
- Row ✅
- Column ✅
- Slider ✅
- Banner (Review comment in file) ❓
- Tabs ✅
- Tab Item ✅

File: packages/pagebuilder/lib/ContentTypes/Column/column.js (hunk 6 above)
Comment (pedrochiossi) [reply packages/pagebuilder/lib/ContentTypes/Column/column.js:227]: ok, fixing it.

File: packages/pagebuilder/lib/ContentTypes/Banner/banner.js (hunk 5 above)
Comment (pedrochiossi) [reply packages/pagebuilder/lib/ContentTypes/Banner/banner.js]: You're right this shouldn't be here, I forgot to remove it.

**Comment by mikhaelbois** [comment]
@pedrochiossi 
1- Just realized that Shimmer components need to be updated to match the min-height received.
2- The Banner component with a background color differs from Luma when appearance is either a Poster or something else.
Here's my proposed solution which might impact the Cypress tests.
```
diff --git a/packages/pagebuilder/lib/ContentTypes/Banner/banner.js b/packages/pagebuilder/lib/ContentTypes/Banner/banner.js
index 848fb184a..e11d94bd4 100644
--- a/packages/pagebuilder/lib/ContentTypes/Banner/banner.js
+++ b/packages/pagebuilder/lib/ContentTypes/Banner/banner.js
@@ -211,7 +211,6 @@ const Banner = props => {
     }, [backgroundElement, image, intersectionObserver, setBgImage]);
--- a/packages/pagebuilder/lib/ContentTypes/Banner/banner.js
+++ b/packages/pagebuilder/lib/ContentTypes/Banner/banner.js
@@ -211,7 +211,6 @@ const Banner = props => {
     }, [backgroundElement, image, intersectionObserver, setBgImage]);

     if (appearance === 'poster') {
-        wrapperStyles.minHeight = minHeight;
         overlayStyles.minHeight = minHeight;
         overlayStyles.paddingTop = paddingTop;
         overlayStyles.paddingRight = paddingRight;
@@ -287,7 +286,7 @@ const Banner = props => {
             className={classes.wrapper}
             style={{
                 ...wrapperStyles,
-                ...mediaQueryStyles
+                ...(appearance !== 'poster' && { ...mediaQueryStyles })
             }}
             ref={backgroundElement}
         >
@@ -296,7 +295,7 @@ const Banner = props => {
                 className={overlayClass}
                 style={{
                     ...overlayStyles,
-                    ...mediaQueryStyles
+                    ...(appearance === 'poster' && { ...mediaQueryStyles })
                 }}
             >
                 <div
```

**Comment by mikhaelbois** [comment]
QA ✅ 
Need https://github.com/magento/pwa-studio/pull/3694 to be finished to make sure Cypress tests are up to date.
